src/apps/app_search.erl

```diff
@@ -190,25 +194,33 @@ search_by_id(TXID) ->
 basic_usage_test() ->
 	ar_storage:clear(),
 	SearchServer = start(),
-	Peers = ar_network:start(10, 10),
-	ar_node:add_peers(hd(Peers), SearchServer),
+	[Peer | _] = ar_network:start(10, 10),
+	ar_node:add_peers(Peer, SearchServer),
 	% Generate the transaction.
 	RawTX = ar_tx:new(),
 	TX = RawTX#tx {tags = [
 		{<<"TestName">>, <<"TestVal">>},
 		{<<"block_height">>, <<"user-specified-block-height">>}
 	]},
-	%% Add tx to network
-	ar_node:add_tx(hd(Peers), TX),
-	%% Begin mining
-	receive after 250 -> ok end,
-	ar_node:mine(hd(Peers)),
-	receive after 1000 -> ok end,
+	AddTx = fun(TXToAdd) ->
+		%% Add tx to network
+		ar_node:add_tx(Peer, TXToAdd),
+		%% Begin mining
+		receive after 250 -> ok end,
+		ar_node:mine(Peer),
+		receive after 1000 -> ok end
+	end,
+	AddTx(TX),
 	%% Get TX by tag
 	TXIDs = get_entries(SearchServer, <<"TestName">>, <<"TestVal">>),
 	?assert(lists:member(TX#tx.id, TXIDs)),
 	%% Get tags by TX
 	{ok, Tags} = get_tags_by_id(SearchServer, TX#tx.id, 3000),
 	?assert(lists:member({<<"TestName">>, <<"TestVal">>}, Tags)),
 	%% Check aux tags
-	?assert({<<"block_height">>, 1} == lists:keyfind(<<"block_height">>, 1, Tags)).
+	?assert({<<"block_height">>, 1} == lists:keyfind(<<"block_height">>, 1, Tags)),
+	%% Check that if writing the TX to the index again, the entries gets
+	%% overwritten, not duplicated.
+	AddTx(TX#tx{tags = [{<<"TestName">>, <<"Updated value">>}]}),
+	{ok, UpdatedTags} = get_tags_by_id(SearchServer, TX#tx.id, 3000),
+	?assertMatch({_, <<"Updated value">>}, lists:keyfind(<<"TestName">>, 1, UpdatedTags)).
```

Comment:
I think, there should be another assertion of the absence of the old value in the list.

Reply:
✅